packages/build/src/steps/get.ts

```diff
@@ -39,7 +39,18 @@ export const getDevSteps = function (command, steps, eventHandlers?: any[]) {
 
   const eventSteps = getEventSteps(eventHandlers)
 
-  const sortedSteps = sortSteps([preDevCleanup, ...steps, eventSteps, devCommandStep], DEV_EVENTS)
+  const sortedSteps = sortSteps(
+    [
+      preDevCleanup,
+      ...steps,
+      // Trigger the uploadBlobs step during development to allow users to test file-based blob
+      // uploads locally and to allow frameworks to capture file-based blobs written by frameworks
+      { ...uploadBlobs, event: 'onDev' },
```

Comment:
I see that you're reusing the core step we have for builds and just changing the `event` property. As far as I know, we don't really have a precedent for running the same core step in multiple events, but I would probably have two separate core steps — one for dev, one for build — that may reuse common functionality. I think taking a core step and creating another one from it on-the-fly here might be confusing to people.

More importantly, if we run the existing core step as is as part of dev, won't we be uploading blobs to the actual production store? How do we ensure that we're uploading blobs to the local store?

Reply:
> I see that you're reusing the core step we have for builds and just changing the `event` property. As far as I know, we don't really have a precedent for running the same core step in multiple events, but I would probably have two separate core steps — one for dev, one for build — that may reuse common functionality. I think taking a core step and creating another one from it on-the-fly here might be confusing to people.

Sure! I can split this into a `dev_blobs_upload` core step that uses shared code.

> More importantly, if we run the existing core step as is as part of dev, won't we be uploading blobs to the actual production store? How do we ensure that we're uploading blobs to the local store?

As I understand it, the answer to your first question is no: The blobs client used in the upload blobs step picks up the local blob server's settings via `process.env['NETLIFY_BLOBS_CONTEXT']`, which is set by both the [dev](https://github.com/netlify/cli/blob/577a6b5cf1c2a26061173af20763227eaeba1ecc/src/commands/dev/dev.ts#L116) and [serve](https://github.com/netlify/cli/blob/577a6b5cf1c2a26061173af20763227eaeba1ecc/src/commands/serve/serve.ts#L101) commands. (`ntl serve` actually already uploads blobs to the local blob server: https://github.com/netlify/cli/pull/6299.)

To your second question: I think this concern would apply equally to `serve`, right? I'm not opposed to adding some checks to ensure `serve` and `dev` don't accidentally upload blobs to production, but given that I think we'd already have seen this issue come up in `serve`, I'm inclined to solve it separately. What do you think?